src/utils/media-devices-manager.js

```diff
@@ -191,6 +200,39 @@ export default class MediaDevicesManager {
     }
   }
 
+  /**
+   * Webkit issue on iOS>13: https://bugs.webkit.org/show_bug.cgi?id=213853
+   */
+  _iOSTrackMutedHack() {
+    const iOSVersion = parseInt(browser.version.split(".")[0]);
+    if (
+      ["iOS", "Mac OS"].includes(detectedOS) &&
+      ["safari", "ios"].includes(browser.name) &&
+      iOSVersion >= IOS_MUTED_HACK_MIN_VERSION
+    ) {
+      this.audioTrack.onmute = this._muteStateChanged;
+      this.audioTrack.onunmute = this._muteStateChanged;
+    }
+  }
+
+  _muteStateChanged() {
+    if (this._audioTrack && this._audioTrack.muted) {
+      if (!this.reconnectIfUnmuted) {
+        this.wasMicEnabledBeforeMute = NAF.connection.adapter.micEnabled;
+        NAF.connection.adapter.enableMicrophone(false);
+      }
+      this.reconnectIfUnmuted = true;
+    } else {
+      if (this.reconnectIfUnmuted) {
+        this.reconnectIfUnmuted = false;
+        this.setMediaStreamToDefault();
```

Comment:
Just to be sure, you probably want 
`await this.setMediaStreamToDefault();` here and add `async` to `_muteStateChanged`